mobile/src/main/java/org/openhab/habdroid/ui/activity/ContentController.java

```diff
@@ -791,6 +793,14 @@ public void onReceivedError(WebView view, int errorCode, String description,
                     updateViewVisibility(true, false);
                 }
             });
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+                WebViewDatabase webViewDatabase = WebViewDatabase.getInstance(getContext());
+                webViewDatabase.setHttpAuthUsernamePassword(Util.getHostFromUrl(url), "",
+                        mConnection.getUsername(), mConnection.getPassword());
+            } else {
+                mWebview.setHttpAuthUsernamePassword(Util.getHostFromUrl(url), "",
+                        mConnection.getUsername(), mConnection.getPassword());
+            }
```

Comment:
I wonder whether it would make sense to put this into a Util method: `void applyAuthentication(WebView WebView, Connection connection)`

Reply:
Done